### DISCUSSION TITLE
cx-Freeze causing circular imports?

### DISCUSSION THREAD

**Discussion by ghost** [marcelotduarte/cx_Freeze#2496] [General]
I've never posted here before, but this time I'm genuinely stuck. I'm using cx_Freeze on a python script that uses pdf2docx. Running the python script normally works fine, but running cx_Freeze's executable gives this error:

```
Traceback (most recent call last):
  File "C:\Users\truma\AppData\Local\Programs\Python\Python312\Lib\site-packages\pymupdf\__init__.py", line 193, in <module>
    from . import mupdf
ImportError: cannot import name 'mupdf' from partially initialized module 'pymupdf' (most likely due to a circular import) (C:\Users\truma\Documents\ballotreaderclone\ballotreader\resources\build\exe.win-amd64-3.12\lib\pymupdf\__init__.pyc)
```

I've tried every suggestion chatgpt had, but nothing worked.

From my analysis it seems like cx_Freeze is messing up the order of imports in all the different python files, and ends up tying itself in a knot that it can't get out of... or something.

I've also tried reinstalling pdf2docx, pymupdf, cx_Freeze, and pip.

To be clear, everything works fine when the script is run from python and even with pyinstaller, but this problem happens with cx_Freeze in particular.

**Comment by marcelotduarte** [comment]
PR #2581 fixes this issue.
